File: hse/BPM161/lupuleac/2/LockFreeSet/src/test/java/ru/hse/lupuleac/lockfree/LinCheckTest.java
package ru.hse.lupuleac.lockfree;

import com.devexperts.dxlab.lincheck.LinChecker;
import com.devexperts.dxlab.lincheck.annotations.Operation;
import com.devexperts.dxlab.lincheck.annotations.Param;
import com.devexperts.dxlab.lincheck.paramgen.IntGen;
import com.devexperts.dxlab.lincheck.strategy.stress.StressCTest;
import org.junit.Test;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

@Param(name = "key", gen = IntGen.class, conf = "1:5")
@StressCTest
public class LinCheckTest {
    private LockFreeSet<Integer> set = new LockFreeSet<>();

    @Operation
    public boolean add(int key) {
        return set.add(key);
    }

    @Operation
    public boolean remove(int key) {
        return set.remove(key);
    }

    @Operation
    public boolean contains(int key) {
        return set.contains(key);
    }

    @Operation
    public boolean isEmpty() {
        return set.isEmpty();
    }

   @Operation
    public String iterator() {
        List<Integer> ints = new ArrayList<>();
        set.iterator().forEachRemaining(ints::add);
        return ints.stream().map(Object::toString).collect(Collectors.joining
                (", "));
    }

    @Test
    public void test() {
        LinChecker.check(LinCheckTest.class);
    }
}

File: hse/BPM161/vasilenko/2/src/test/kotlin/lockfree/LinCheckTest.kt
package lockfree

import com.devexperts.dxlab.lincheck.LinChecker
import com.devexperts.dxlab.lincheck.annotations.Operation
import com.devexperts.dxlab.lincheck.annotations.Param
import com.devexperts.dxlab.lincheck.paramgen.IntGen
import com.devexperts.dxlab.lincheck.strategy.stress.StressCTest
import org.junit.Test

@StressCTest(threads = 3, actorsBefore = 3, actorsAfter = 3)
class LinCheckTest {
    private val set = LockFreeSet<Int>()

    @Operation
    fun add(@Param(gen = IntGen::class) item: Int): Boolean {
        return set.add(item)
    }

    @Operation
    fun remove(@Param(gen = IntGen::class) item: Int): Boolean {
        return set.remove(item)
    }

    @Operation
    fun isEmpty(): Boolean {
        return set.isEmpty
    }

    @Operation
    fun contains(@Param(gen = IntGen::class) item: Int): Boolean {
        return set.contains(item)
    }

    @Operation
    fun iterator(): List<Int> {
        return set.iterator().asSequence().toList()
    }

    @Test
    fun test() {
        LinChecker.check(LinCheckTest::class.java)
    }
}